pkg/storage/fs/posix/blobstore/blobstore.go

```diff
@@ -63,15 +65,35 @@ func (bs *Blobstore) Upload(node *node.Node, source string) error {
 	if err != nil {
 		return errors.Wrapf(err, "could not write blob '%s'", node.InternalPath())
 	}
-
 	err = w.Flush()
 	if err != nil {
 		return err
 	}
+	err = os.Chtimes(path, fi.ModTime(), fi.ModTime())
+	if err != nil {
+		return err
+	}
 
-	if fi != nil {
-		return os.Chtimes(path, fi.ModTime(), fi.ModTime())
+	if copyTarget != "" {
+		// also "upload" the file to a local path, e.g. for keeping the "current" version of the file
+		err := os.MkdirAll(filepath.Dir(copyTarget), 0700)
+		if err != nil {
+			return err
+		}
+
+		file.Seek(0, 0)
+		copyFile, err := os.OpenFile(copyTarget, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0700)
```

Comment:
Why the `0700` permission? I guess you want `0600` here? Or even the same as the source file?

Reply:
The revision blobs aren't directly used as files, but are read and overwrite the target when they are restored so I don't think we have to bother about the permissions of the actual file (which could also change after creating the revision). Changing it to 0600.